### PR TITLE
Save vectors as little endian, load with Ops.asarray

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

* Always save vector data as little endian
* Always run `Vectors.to_ops` when vector data is loaded so that `Ops.asarray` can be used to load the data correctly for the current ops.

### Types of change

?

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
This requires https://github.com/explosion/thinc/pull/584 and updated thinc requirements.

File: spacy/vectors.pyx [hunk 1]
```diff
@@ -629,7 +631,8 @@ cdef class Vectors:
             if hasattr(self.data, "to_bytes"):
                 return self.data.to_bytes()
             else:
-                return srsly.msgpack_dumps(self.data)
+                ops = get_current_ops()
+                return srsly.msgpack_dumps(ops.to_numpy(self.data, byteorder="<"))
```
Comment (svlandeg) [inline spacy/vectors.pyx:635]:
```suggestion
                return srsly.msgpack_dumps(ops.to_numpy(self.data, byte_order="<"))
```

File: spacy/vectors.pyx [hunk 2]
```diff
@@ -563,8 +563,9 @@ cdef class Vectors:
             # the source of numpy.save indicates that the file object is closed after use.
             # but it seems that somehow this does not happen, as ResourceWarnings are raised here.
             # in order to not rely on this, wrap in context manager.
+            ops = get_current_ops()
             with path.open("wb") as _file:
-                save_array(self.data, _file)
+                save_array(ops.to_numpy(self.data, byteorder="<"), _file)
```
Comment (svlandeg) [inline spacy/vectors.pyx:568]:
```suggestion
                save_array(ops.to_numpy(self.data, byte_order="<"), _file)
```

**Comment by svlandeg** [comment]
Should this PR bump the Thinc requirements, or how would you handle that?

**Comment by adrianeboyd** [comment]
I've also tested with the trained pipeline training and didn't run into any issues.

**Pull request merged**